### PR TITLE
Added documentation for vehicle connectivity

### PR DESCRIPTION
**Pull request by ThomasAlxDmy**
# Description

More friendly documentation & Added docs for vehicle connectivity

Fixes # (issue)

## Type of change

Please select all options that apply to this change:

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [x] My code follows the style of this project.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding updates to the documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,8 +147,8 @@ spec:
 
 Vehicles must be running firmware version 2023.20.6 or later.  Some older model S/X are not supported.
 
-## Backends/dispatchers
-The following [dispatchers](./telemetry/producer.go#L10-L19) are supported
+## Personalized Backends/Dispatchers
+Dispatchers are the mechanism fleet_telemtetry server will push data to. They can of any type from distributed message queues to single STDOUT logger. Here's a list of currently supported [dispatchers](./telemetry/producer.go#L10-L19):
```
Comment (agbpatro) [inline README.md:151]: nit:-
```md
Dispatchers are the mechanisms by which the fleet_telemetry server will push data. They can be of any type, from distributed message queues to a single STDOUT logger.  Here is a list of the currently supported [dispatchers](./telemetry/producer.go#L10-L19):
```

File: README.md [hunk 2]
```diff
@@ -156,12 +158,23 @@ The following [dispatchers](./telemetry/producer.go#L10-L19) are supported
...
 
 ## Reliable Acks
 Fleet telemetry can send ack messages back to the vehicle. This is useful for applications that need to ensure the data was received and processed. To enable this feature, set `reliable_ack_sources` to one of configured dispatchers (`kafka`,`kinesis`,`pubsub`,`zmq`) in the config file. Reliable acks can only be set to one dispatcher per recordType. See [here](./test/integration/config.json#L8) for sample config.
 
+## Detecting Vehicle Connectivity Changes
+On the vehicle, Fleet telemetry client behave similarly to how the connectivity engine for vehicle commands. Therefore we can use Fleet telemetry connectivity event to assume when a vehicle is online. Note that it is a proxy, but if configured properly fleet telemetry connectivity time should match vehicle connectivity state in 99%+. To enable connectivity events simply add the `connectivity` records in the list of events in config.json file:
+
+  ```
+    "records": {
```
Comment (agbpatro) [inline README.md:171]: nit: maybe refer to integration test config file?
https://github.com/teslamotors/fleet-telemetry/blob/main/test/integration/config.json#L49

**Review by agbpatro — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -5,9 +5,11 @@
...
+Fleet Telemetry is a server reference implementation for Tesla's telemetry protocol. It is the best way to get data from Tesla vehicles at regular intervals. Owners can allow registered applications to receive telemetry securely and directly from their vehicles. This reference implementation can be used by individual owners as is or by fleet operators who can extend it to aggregate data accross their fleet.
 
-The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses.
+The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses. The application also provide a proxy interface to know if vehicle are online or not.
```
Comment (aaronpkahn) [inline README.md:10]: Recommended change: `The application also provide a proxy interface to know if vehicle are online or not` to something like: `The application also provides events to determine the vehicle connection state, which can be used as a proxy for vehicle online state.`

Comment (sethterashima) [inline README.md:10]: "also provide" -> "also provides"
"vehicle are" -> "vehicles are" or "a vehicle is"
"to know if" -> I think "learn" or "determine" is better here.
"or not" is redundant.

So:

The application also provides an interface for determining if a vehicle is online.

File: README.md (hunk 1 above)
Comment (sethterashima) [reply README.md:151]: Or "Fleet Telemetry servers push data using dispatchers."

File: README.md [hunk 4]
```diff
@@ -156,12 +158,23 @@ The following [dispatchers](./telemetry/producer.go#L10-L19) are supported
...
+
 >NOTE: To add a new dispatcher, please provide integration tests and updated documentation. To serialize dispatcher data as json instead of protobufs, add a config `transmit_decoded_records` and set value to `true` as shown [here](config/test_configs_test.go#L186)
 
 ## Reliable Acks
 Fleet telemetry can send ack messages back to the vehicle. This is useful for applications that need to ensure the data was received and processed. To enable this feature, set `reliable_ack_sources` to one of configured dispatchers (`kafka`,`kinesis`,`pubsub`,`zmq`) in the config file. Reliable acks can only be set to one dispatcher per recordType. See [here](./test/integration/config.json#L8) for sample config.
 
+## Detecting Vehicle Connectivity Changes
+On the vehicle, Fleet telemetry client behave similarly to how the connectivity engine for vehicle commands. Therefore we can use Fleet telemetry connectivity event to assume when a vehicle is online. Note that it is a proxy, but if configured properly fleet telemetry connectivity time should match vehicle connectivity state in 99%+. To enable connectivity events simply add the `connectivity` records in the list of events in config.json file:
```
Comment (sethterashima) [inline README.md:168]: Fleet Telemetry is not consistently capitalized in this file.

File: README.md [hunk 5]
```diff
@@ -5,9 +5,11 @@
...
 
-The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses.
+The service handles device connectivity as well as receiving and storing transmitted data. Once configured, devices establish a WebSocket connection to push configurable telemetry records. Fleet Telemetry provides clients with ack, error, or rate limit responses. The application also provides an event stream to determine the vehicle connection state, which can be used as a proxy for vehicle online state.
+
+By configuring `fleet_telemetry_config`, individual owners and fleet operators can easily control the timing and conditions of vehicle data transmission (live / on change / at frenquency).
```
Comment (agbpatro) [inline README.md:12]: maybe link to documentation?
https://developer.tesla.com/docs/fleet-api/endpoints/vehicle-endpoints#fleet-telemetry-config-jws

**Pull request merged**